### ISSUE TITLE
gateway with two different protocol [HTTP & HTTPS] and same port 443 ,listeners are not generated correctly

### ISSUE THREAD

**Issue by PayalKhade** [istio/istio#48078]
### Is this the right place to submit this?

- [X] This is not a security vulnerability or a crashing bug
- [X] This is not a question about how to use Istio

### Bug Description

We have two different gateway , which was using same port 443 but different protocol. 
When we try to upgrade istio from1.16.2 --> 1.18.2 we faced issuw for ingressgateway pod.

istio-ingress pod not in 0/1 running state .
pod description:
Warning  Unhealthy  4m55s (x464 over 19m)  kubelet            Readiness probe failed: HTTP probe failed with statuscode: 503
Pod log
warn    Envoy proxy is NOT ready: config received from XDS server, but was rejected: cds updates: 2 successful, 0 rejected; lds updates: 0 successful, 2 rejected

we found below logs for istiod.
`Searching for: istio-ingressgateway-5b6d6cbd9f-mmxwx.istio-system-6 Internal:Error adding/updating listener(s) 0.0.0.0_8443: error adding listener \'0.0.0.0:8443\': filter chain \'\' has the same matching rules defined as \'\'`

Please find gateway ss below
<img width="591" alt="image" src="https://github.com/istio/istio/assets/51352754/2a035ac4-6df5-4463-9f01-85b1483104ef">
<img width="381" alt="image" src="https://github.com/istio/istio/assets/51352754/bc6424ee-e261-4135-ab54-85b901ad578c">

We don't face any issue in istio version 1.16.2 .

### Version

```Text
payalkhade@INPHWN154192:~$ istioctl version --remote
client version: 1.20.0
control plane version: 1.18.2
data plane version: 1.18.2 (57 proxies)
```

### Additional Information

We updated port in ipg gateway from 443 -->80 , ingressgateway pods comes in running state without any isssue.

**Comment by howardjohn** [comment]
I would imagine this is caused by other configuration that you haven't included. Can you show all the Gateway, VirtualService, and Service? or `istioctl bug-report`

**Comment by PayalKhade** [comment]
We just upgraded istio from 1.16.2 -- 1.18.2 , didn't add/remove any changes. Other than upgrade changes.
In 1.16.2 , we had 2 gateways and still everything working well.
In 1.18.2, we faced issue which i mentioned earlier.

We also tried to upgrade every component of istio to version 1.18.2 except istiod (istio discovery) . we kept istiod version to 1.16.2 only, that time also we didn't face any issue. Ingressgateway pods were also in running state.

When we changed port no. from ipg gateway & updated istiod to 1.18.2 that issue got resolved.

Gateway : We had 2 gateways only , which shared earlier.
VirtualService: 
<img width="414" alt="image" src="https://github.com/istio/istio/assets/51352754/4b46110c-f825-4470-a242-57d085a4a30e">
<img width="336" alt="image" src="https://github.com/istio/istio/assets/51352754/1e33c82b-7ccb-4e6c-8c8b-f6d8986075c5">

config _dump file. 
[config_dump_notrun.json](https://github.com/istio/istio/files/13503221/config_dump_notrun.json)

**Comment by howardjohn** [comment]
No I mean your config is definitelyi nvalid on all versions, its just failing in a different way on 1.18.

VS1 and VS2 are both listening on port 443 for `*` (any) host. So they are indistinguishable. Probably in older version, one was silently winning.

I vaguely recall we already fixed this, so 1.20 may behave like 1.16

**Comment by PayalKhade** [comment]
Can you please explain more about this?
`I vaguely recall we already fixed this, so 1.20 may behave like 1.16`
What has fixed ? and what exactly meaning of 1.20 may behave like 1.16 ?

Also, I have question regarding istio support
When we say istio 1.16.x Supported Kubernetes Versions [1.22, 1.23, 1.24, 1.25], what exactly it means, if i use 1.16.x version for k8 version 1.27 will it break?
